### PR TITLE
Topic/hostfile fraction of nodes

### PR DESCRIPTION
**Pull request by ribab**
Hostfile to dynamically generate list of nodes

This list can be a fraction of all nodes, or a fraction of a subset

**Merging this will also merge the fix for ExecuteCmd #649 (if not already merged)

### CODE REVIEW

File: pylib/Utilities/ExecuteCmd.py [hunk 1]
```diff
@@ -110,24 +110,32 @@ def execute(self, options, cmdargs, testDef):
...
-                        read = p.stderr.readline().rstrip()
+                        read = p.stderr.readline()
                         if read:
+                            read = read.rstrip()
                             testDef.logger.verbose_print('stderr: ' + read.decode("utf-8", errors='replace'))
                             stderr.append(read.decode("utf-8", errors='replace'))
+                            stderr_done = False
 
-                if p.poll() != None:
+                if stdout_done and stderr_done:
```
Comment (rhc54) [inline pylib/Utilities/ExecuteCmd.py]: I'm probably just missing it, but I'm not sure I understand this completion test. If I read either stdout or stderr, then you set the corresponding flag to False - which means I will exit this loop. But that means I won't have read the other fd should something have come out that way. Is that the intended behavior - i.e., you only wait for the first one to get output and then quit?

Comment (rhc54) [reply pylib/Utilities/ExecuteCmd.py]: Oh wait - it's the other way around. You only exit if neither one has something to read. I gather that p.stdout.readline and p.stderr.readline are blocking calls then?

Comment (rhc54) [reply pylib/Utilities/ExecuteCmd.py]: Guess what I'm saying is that I'm having a real problem understanding this logic. I gather that the p.poll test was a problem somehow??

Comment (ribab) [reply pylib/Utilities/ExecuteCmd.py]: Yes, the p.poll test was a problem.
I tried using a debugger to see what p.poll returns even before all output was read using readline(), and it returns 0 in the debugger

Comment (ribab) [reply pylib/Utilities/ExecuteCmd.py]: I only quit if both stdout and stderr have no output (not even a \n newline as output)

Comment (rhc54) [reply pylib/Utilities/ExecuteCmd.py]: ok - thanks!

**Pull request merged**